src/qgis_geonode/gui/geonode_source_select_provider.py

```diff
@@ -542,4 +558,106 @@ def update_keywords(self, keywords: typing.Optional[typing.List[str]] = None):
+
+    def restore_settings(self):
+        current_search_filters = settings_manager.get_current_search_filters()
+        # if keywords list exist populate the keywords list first
+        keywords = current_search_filters.keywords
+        if keywords is not None:
+            self.keyword_cmb.addItem("")
+            self.keyword_cmb.addItems(keywords)
+        if current_search_filters.title is not None:
+            self.title_le.setText(current_search_filters.title)
+        if current_search_filters.abstract is not None:
+            self.abstract_le.setText(current_search_filters.abstract)
+        if current_search_filters.keyword is not None:
+            index = self.keyword_cmb.findText(current_search_filters.keyword)
+            self.keyword_cmb.setCurrentIndex(index)
+        if current_search_filters.topic_category is not None:
+            index = self.category_cmb.findText(current_search_filters.topic_category)
+            self.category_cmb.setCurrentIndex(index)
+        if current_search_filters.temporal_extent_start is not None:
+            self.temporal_extent_start_dte.setDateTime(
+                current_search_filters.temporal_extent_start
+            )
+        if current_search_filters.temporal_extent_end is not None:
+            self.temporal_extent_end_dte.setDateTime(
+                current_search_filters.temporal_extent_end
+            )
+        if current_search_filters.publication_date_start is not None:
+            self.publication_start_dte.setDateTime(
+                current_search_filters.publication_date_start
+            )
+        if current_search_filters.publication_date_end is not None:
+            self.publication_end_dte.setDateTime(
+                current_search_filters.publication_date_end
+            )
+        if current_search_filters.spatial_extent is not None:
+            self.spatial_extent_box.setOutputExtentFromUser(
+                current_search_filters.spatial_extent,
+                qgis.core.QgsCoordinateReferenceSystem("EPSG:4326"),
+            )
+        self.vector_chb.setChecked(
+            (
+                models.GeonodeResourceType.VECTOR_LAYER
+                in current_search_filters.layer_types
+            )
+        )
+        self.raster_chb.setChecked(
+            (
+                models.GeonodeResourceType.RASTER_LAYER
+                in current_search_filters.layer_types
+            )
+        )
+        self.map_chb.setChecked(
+            (models.GeonodeResourceType.MAP in current_search_filters.layer_types)
+        )
+        # trigger actions when resource types buttons have been toggled
+        self.resource_types_btngrp.buttonClicked.emit(None)
+        sort_index = self.sort_field_cmb.findData(current_search_filters.ordering_field)
+        self.sort_field_cmb.setCurrentIndex(sort_index)
+
+        self.reverse_order_chb.setChecked(current_search_filters.reverse_ordering)
+
+    def save_settings(self):
```

Comment:
As mentioned above, please use a more helpful name for this method

```suggestion
    def save_search_filters(self):
```